app/shop.py

```diff
@@ -0,0 +1,44 @@
+from dataclasses import dataclass
+from math import sqrt
+from decimal import getcontext, Decimal
+
+
+@dataclass
+class Shop:
+    name: str
+    location: list
+    products: dict
+
+    def calc_product_price(
+            self,
+            customer_cart: dict,
+            product: str
+    ) -> float:
+        cost = customer_cart[product] * self.products[product]
+        if cost == int(cost):
+            cost = int(cost)
+        return cost
+
+    def calc_cart_price(self, customer_cart: dict) -> float:
+        total = [
+            self.calc_product_price(customer_cart, product)
+            for product in customer_cart
+        ]
+        return sum(total)
+
+    def calc_trip_price(
```

Comment:
Do not use shortening in your method names, there is no need for it. Use full `calculate` in your method names.

Reply:
This comment was not implemented, rename your method names to be more descriptive and without the use of shortenings.